airbyte-integrations/bases/base-java-s3/src/test/java/io/airbyte/integrations/destination/s3/S3BaseChecksTest.java

```diff
@@ -49,4 +52,45 @@ public void attemptWriteAndDeleteS3Object_should_createSpecificFiles() {
     verify(s3Client).deleteObject(eq("test_bucket"), startsWith("test/bucket/path/_airbyte_connection_test_"));
   }
 
+  @Test
+  public void attemptWriteAndDeleteS3Object_should_skipDirectoryCreateIfRootPath() {
+    S3DestinationConfig config = new S3DestinationConfig(
+        null,
+        "test_bucket",
+        "",
+        null,
+        null,
+        null,
+        null,
+        s3Client);
+    S3StorageOperations operations = new S3StorageOperations(new S3NameTransformer(), s3Client, config);
+
+    S3BaseChecks.attemptS3WriteAndDelete(operations, config, "");
+
+    verify(s3Client, never()).putObject("test_bucket", "", "");
+    verify(s3Client).putObject(eq("test_bucket"), startsWith("_airbyte_connection_test_"), anyString());
+    verify(s3Client).listObjects(ArgumentMatchers.<ListObjectsRequest>argThat(request -> "test_bucket".equals(request.getBucketName())));
+    verify(s3Client).deleteObject(eq("test_bucket"), startsWith("_airbyte_connection_test_"));
+  }
+
+  @Test
+  public void attemptWriteAndDeleteS3Object_should_skipDirectoryCreateIfNullPath() {
```

Comment:
Looks good overall, just noticed that there's a seemingly random `_` in the method name and in the existing methods (so presume you're keeping with what's already existing). Minor nit but to keep test naming conventions, would it make more sense to have this consistent with camelCase?

Reply:
lol this is actually my preferred testcase naming convention - imo the `testWhateverMethod` style is harder to read and doesn't provide much info about what the testcase is intended to do

iirc there's a few of these scattered through the codebase (which is where I picked it up from). Not strongly opinionated on this but mildly prefer to spread this style further.